applications/neuroscan/frontend/src/redux/middleware.js

```diff
@@ -186,34 +218,19 @@ const middleware = (store) => (next) => (action) => {
               next(raiseError(msg));
             });
         } else {
-          const neurons = getInstancesOfType(addedObjectsToViewer, NEURON_TYPE) || ['-1'];
-          const contacts = getInstancesOfType(addedObjectsToViewer, CONTACT_TYPE) || ['-1'];
-          const synapses = getInstancesOfType(addedObjectsToViewer, SYNAPSE_TYPE) || ['-1'];
+          const neurons = getInstancesOfType(addedObjectsToViewer, NEURON_TYPE) || [];
+          const contacts = getInstancesOfType(addedObjectsToViewer, CONTACT_TYPE) || [];
+          const synapses = getInstancesOfType(addedObjectsToViewer, SYNAPSE_TYPE) || [];
 
-          neuronService.getByUID(timePoint, neurons.map((n) => n.uidFromDb))
-            .then((newNeurons) => {
-              contactService.getByUID(timePoint, contacts.map((n) => n.uidFromDb))
-                .then((newContacts) => {
-                  synapseService.getByUID(timePoint, synapses.map((n) => n.uidFromDb))
-                    .then((newSynapses) => {
-                      const newInstances = newNeurons.concat(newContacts.concat(newSynapses))
-                        .map((i) => mapToInstance(i));
-                      widget.config.timePoint = timePoint; // update the current widget's timepoint
-                      createSimpleInstancesFromInstances(newInstances)
-                        .then(() => {
-                          store
-                            .dispatch(
-                              addToWidget(
-                                widget,
-                                newInstances,
-                                true,
-                                addedObjectsToViewer,
-                              ),
-                            );
-                        });
-                    });
-                });
-            });
+          const newNeurons = await fetchDataForEntity(neuronService, timePoint, neurons);
+          const newContacts = await fetchDataForEntity(contactService, timePoint, contacts);
+          const newSynapses = await fetchDataForEntity(synapseService, timePoint, synapses);
+
+          const newInstances = [...newNeurons, ...newContacts, ...newSynapses]
+            .map((i) => mapToInstance(i));
+          widget.config.timePoint = timePoint; // update the current widget's timepoint
```

Comment:
This line is missing in the CPHATE case:

See https://github.com/MetaCell/geppetto-NeuroSCAN/blob/85f381331e74b012bde7479f0714687752be06b2/applications/neuroscan/frontend/src/redux/middleware.js#L177C1-L178C1

Reply:
nice catch @afonsobspinto , I've added the line